src/ObjectPool/src/ObjectPoolServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.ObjectPool;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+/// <summary>
+/// Extension methods for adding <see cref="ObjectPool{T}"/> to DI container.
+/// </summary>
+public static class ObjectPoolServiceCollectionExtensions
+{
+    /// <summary>
+    /// Adds an <see cref="ObjectPool{T}"/> and lets DI return scoped instances of T.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add to.</param>
+    /// <param name="configureOptions">The action used to configure the options of the pool.</param>
+    /// <typeparam name="TService">The type of objects to pool.</typeparam>
+    /// <returns>Provided service collection.</returns>
+    /// <exception cref="ArgumentNullException">When <paramref name="services"/> is <see langword="null"/>.</exception>
+    /// <remarks>
+    /// The default capacity is <c>Environment.ProcessorCount * 2</c>.
+    /// The pooled type instances are obtainable the same way like any other type instances from the DI container.
```

Comment:
Is this line really true? Don't I need to get the `ObjectPool<TService>` from the DI container, and then get the pooled type instances?